### PR TITLE
Dsc ci cd

### PR DESCRIPTION
**Pull request by eslesar**
@grayzu Please review. This should not be merged until https://github.com/PowerShell/Demo_CI/pull/6 is merged.

@zjalexander Can you also please review?

### CODE REVIEW

**Comment by saldana** [comment]
### :warning: Validation status: warnings


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[dsc/dscCiCd.md](https://github.com/PowerShell/PowerShell-Docs/blob/dsc-ci-cd/dsc/dscCiCd.md) | :warning:Warning | | [Details](#user-content-dsc%2FdscCiCd.md)
[dsc/TOC.md](https://github.com/PowerShell/PowerShell-Docs/blob/dsc-ci-cd/dsc/TOC.md) | :white_check_mark:Succeeded | |
[dsc/dscExamples.md](https://github.com/PowerShell/PowerShell-Docs/blob/dsc-ci-cd/dsc/dscExamples.md) | :white_check_mark:Succeeded | |

<a name="dsc%2FdscCiCd.md"></a>
### [dsc/dscCiCd.md](https://github.com/PowerShell/PowerShell-Docs/blob/dsc-ci-cd/dsc/dscCiCd.md)
 - **[Warning]** Extra newline detected: &quot;`\Assets\DscPipelineTools\DscPipelineTools.psm1) 
programatically creates a configuration data document from the hashtable (node data) and array (non-node data) 
that are passed as the `&quot;. Please remove the extra newline.

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C4%5C6%5Cbab27324-4c59-a348-1473-9c6d587e4020%5CPullRequest%5C201704062355105196-1142%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by saldana** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[dsc/TOC.md](https://github.com/PowerShell/PowerShell-Docs/blob/dsc-ci-cd/dsc/TOC.md) | :white_check_mark:Succeeded | |
[dsc/dscCiCd.md](https://github.com/PowerShell/PowerShell-Docs/blob/dsc-ci-cd/dsc/dscCiCd.md) | :white_check_mark:Succeeded | |
[dsc/dscExamples.md](https://github.com/PowerShell/PowerShell-Docs/blob/dsc-ci-cd/dsc/dscExamples.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C4%5C7%5Cbab27324-4c59-a348-1473-9c6d587e4020%5CPullRequest%5C201704070001142865-1142%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: dsc/TOC.md [hunk 1]
```diff
@@ -54,6 +54,10 @@
 ## [Using a DSC report server](reportServer.md)
 ## [Pull server best practices](secureServer.md)
 
+# [DSC examples](dscExamples.md)
+## [Building a CI-CD pipeline with DSC, Pester, and Visual Studio Team Services](dscCiCd.md)
```
Comment (zjalexander) [inline dsc/TOC.md]: Would it be better to spell out the CI-CD acronym and remove "...with DSC, Pester, and Visual studio..." from the title of this article?

File: dsc/dscCiCd.md [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+---
+
+# Building a CI-CD pipeline with DSC, Pester, and Visual Studio Team Services
+
+This example demonstrates how to build a Continuous Integration-Continuous Deployment (CI-CD) pipeline by using PowerShell,
+DSC, Pester, and Visual Studio Team Foundation Server (TFS).
+
+After the pipeline is built and configured, you can use it to fully deploy, configure and test a DNS server and associated host records. 
+This process simulates the first part of a pipeline that would be used in a development environment.
+
```
Comment (zjalexander) [inline dsc/dscCiCd.md]: I think this is missing a section on why this is a good idea, the benefits associated with it, etc.

File: dsc/TOC.md (hunk 1 above)
Comment (eslesar) [inline dsc/TOC.md]: I'm open to expanding the acronyms. I'd at least like to keep DSC in the title for SEO reasons, and so people understand that this is DSC-related when they hit it. There are a lot of topics on CI/CD out there.

Comment (ArieHein) [inline dsc/TOC.md]: If you must use acronyms, then use CI/CD(as @eslesar mentioned), not CI-CD.
The first is widely used in the field, the later has no "meaning" :)

**Comment by saldana** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[dsc/TOC.md](https://github.com/PowerShell/PowerShell-Docs/blob/dsc-ci-cd/dsc/TOC.md) | :white_check_mark:Succeeded | |
[dsc/dscCiCd.md](https://github.com/PowerShell/PowerShell-Docs/blob/dsc-ci-cd/dsc/dscCiCd.md) | :white_check_mark:Succeeded | |
[dsc/dscExamples.md](https://github.com/PowerShell/PowerShell-Docs/blob/dsc-ci-cd/dsc/dscExamples.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C6%5C2%5Cbab27324-4c59-a348-1473-9c6d587e4020%5CPullRequest%5C201706021834331918-1142%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**